File: backend/src/models/Escalation.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IEscalation extends Document {
  customerId: mongoose.Types.ObjectId;
  reason: string;
  status: 'Pending' | 'Resolved' | 'In Progress';
  severity: 'Low' | 'Medium' | 'High' | 'Critical';
  notes?: string;
  assignedTo?: mongoose.Types.ObjectId;
  createdAt: Date;
  updatedAt: Date;
}

const EscalationSchema: Schema = new Schema(
  {
    customerId: { type: Schema.Types.ObjectId, ref: 'Customer', required: true, index: true },
    reason: { type: String, required: true },
    status: { type: String, enum: ['Pending', 'Resolved', 'In Progress'], default: 'Pending', index: true },
    severity: { type: String, enum: ['Low', 'Medium', 'High', 'Critical'], default: 'Medium', index: true },
    notes: { type: String },
    assignedTo: { type: Schema.Types.ObjectId, ref: 'User' },
  },
  { timestamps: true }
);

export default mongoose.model<IEscalation>('Escalation', EscalationSchema);

File: backend/models/escalation.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class EscalationCreate(BaseModel):
    customer_id: str
    reason: str
    severity: str = "High"  # Low | Medium | High | Critical
    status: str = "Pending"  # Pending | In Progress | Resolved
    assigned_to: Optional[str] = None


class EscalationResponse(EscalationCreate):
    id: str
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
